datafusion/core/src/datasource/listing/table.rs

```diff
@@ -817,6 +817,20 @@ impl TableProvider for ListingTable {
             .map(|col| Ok(self.table_schema.field_with_name(&col.0)?.clone()))
             .collect::<Result<Vec<_>>>()?;
 
+        // If the filters can be resolved using only partition cols, there is no need to
+        // pushdown it to TableScan, otherwise, `unhandled` pruning predicates will be generated
+        let table_partition_col_names = table_partition_cols
```

Comment:
This makes sense -- but I don't see a test for it. I also feel like this logic might be duplicated elsewhere(maybe the "can push filters code")? 

What would you think about updating the fields on `ListingTableProvider` somehow to reflect this knowledge a bit more centrally?

Reply:
I conducted some tests and discovered that filters that can be resolved using only `table_partition_cols` generate `unhandled` predicates when creating `PruningPredicate`,  which causes `statistics()` to be unable to generate exact stats in some cases.

Reply:
`filters` are passed as parameters to `ListingTable::scan`, so I believe storing `table_partition_col_names` or `filters` in ListingTable has no practical use. Additionally, I think the correctness of this change can be covered by existing tests, such as those in `datafusion/core/tests/sql/path_partition.rs`.

Reply:
Thank you for your suggestion, I've added a new test named `parquet_partition_pruning_filter` in `datafusion/core/tests/sql/path_partition.rs`.